scipy/signal/tests/test_signaltools.py

```diff
@@ -1093,19 +1093,17 @@ def test_types(self, dtype):
         assert_equal(signal.medfilt(in_typed).dtype, dtype)
         assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
 
-    def test_types_deprecated(self):
-        dtype = np.longdouble
-        in_typed = np.array(self.IN, dtype=dtype)
-        msg = "Using medfilt with arrays of dtype"
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt(in_typed).dtype, dtype)
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
-
-
     @pytest.mark.parametrize('dtype', [np.bool_, np.complex64, np.complex128,
-                                       np.clongdouble, np.float16,])
+                                       np.clongdouble, np.float16, np.object_,
+                                       "float96", "float128"])
     def test_invalid_dtypes(self, dtype):
+        # Only want to run the test for `float96` and `float128` if the platform has
+        # them. We check what the largest dtype is and compare to see if it is the
+        # requested `float96`, `float128`, or niether.
+        if (dtype in ["float96", "float128"]
+            and np.finfo(np.longdouble).dtype != dtype):
+            pytest.skip(f"Platform does not support {dtype}")
```

Comment:
> unfortunately the suggestion won't work as a platform will only have float96 or float128 so we still need to skip one of them even if `np.finfo(np.longdouble).dtype.itemsize > 12`

I still don't understand what the second half here is supposed to achieve. I'm guessing the dtype `==` is overloaded to return `True` if the string matches the underlying dtype.

I also don't understand your point about why it would fail for one or the other if we use my suggestion based on `.itemsize`. Since we're now skipping unconditionally if the dtype is not in `float32` / `float64`, any `longdouble` that's at least 12 bytes long should be rejected.

Reply:
> > unfortunately the suggestion won't work as a platform will only have float96 or float128 so we still need to skip one of them even if `np.finfo(np.longdouble).dtype.itemsize > 12`
> 
> I still don't understand what the second half here is supposed to achieve. I'm guessing the dtype `==` is overloaded to return `True` if the string matches the underlying type.

Yes so for example on my platform:
```
In [2]: np.finfo(np.longdouble).dtype == "float96"
Out[2]: False

In [3]: np.finfo(np.longdouble).dtype == "float128"
Out[3]: True
```
so the test will run for `float128` and be skipped for `float96` as desired.

> I also don't understand your point about why it would fail for one or the other if we use my suggestion based on `.itemsize`. Since we're now skipping unconditionally if the dtype is not in `float32` / `float64`, any `longdouble` that's at least 12 bytes long should be rejected.

I am not sure follow as the other condition is `dtype in ["float96", "float128"]` not `float32` / `float64`

```
In [6]: np.finfo(np.longdouble).dtype.itemsize < 12
Out[6]: False

In [7]: np.float96
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[7], line 1
----> 1 np.float96

File ~/miniconda3/envs/scipy-dev/lib/python3.12/site-packages/numpy/__init__.py:333, in __getattr__(attr)
    330     "Removed in NumPy 1.25.0"
    331     raise RuntimeError("Tester was removed in NumPy 1.25.")
--> 333 raise AttributeError("module {!r} has no attribute "
    334                      "{!r}".format(__name__, attr))

AttributeError: module 'numpy' has no attribute 'float96'
```

Reply:
OK, I get it now. Here's my suggestion
```suggestion
        # We can only test this on platforms that support a native type of float96 or
        # float128; comparing to np.longdouble allows us to filter out non-native types
        if (dtype in ["float96", "float128"]
                and np.finfo(np.longdouble).dtype != dtype):
            pytest.skip(f"Platform does not support {dtype}")
```

Reply:
Thanks, have committed your sugestion